src/main.rs

```diff
@@ -109,6 +146,37 @@ fn main() -> Result<()> {
     let mut buffer = Vec::new();
     input.read_to_end(&mut buffer)?;
 
+    let schema_string = opts.read_schema_to_string().unwrap_or_else(|e| {
+        eprintln!("Failed to read schema: {}", e);
+        String::new()
+    });
+
+    let query_string = opts.read_query_to_string().unwrap_or_else(|e| {
+        eprintln!("Failed to read query: {}", e);
+        String::new()
+    });
+
+    let scale_factor = if !schema_string.is_empty() && !query_string.is_empty() {
+        let input_json: serde_json::Value = match serde_json::from_slice(&buffer) {
+            Ok(json) => json,
+            Err(e) => {
+                eprintln!("Failed to parse input as JSON: {}", e);
+                bail!("Invalid input JSON: {}", e)
+            }
+        };
+
+        BluejaySchemaAnalyzer::analyze_schema_definition(&schema_string, &query_string, &input_json)
+            .unwrap_or_else(|e| {
+                eprintln!("Error analyzing schema: {}", e);
+                eprintln!("Default resource limits will be used.");
+                1.0 // Use default scale factor on error
+            })
+    } else {
+        eprintln!("Analysis skipped due to missing schema or query.");
+        eprintln!("Default resource limits will be used.");
```

Comment:
Is this something that should be printed to stderr? Do we want to print anything at all?

Reply:
Hmm, originally was thinking yes, but it seems this message would get lost in the output anyways. Also, I dont think the partners needs to know this. Since with CLI changes, there should be no updates or changes in DX from their end - i dont think this message is usefull.